src/algorithm/contains.rs

```diff
@@ -0,0 +1,277 @@
+        }
+        // check if point is a vertex
+        if vect.contains(p) {
+            return true;
+        }
+        for (p1, p2) in vect.iter().zip(vect[1..].iter()) {
+            if ((p1.lat() == p2.lat()) && (p1.lat() == p.lat()) &&
+                (p.lng() > p1.lng().min(p2.lng())) &&
+                (p.lng() < p1.lng().max(p2.lng()))) ||
+               ((p1.lng() == p2.lng()) && (p1.lng() == p.lng()) &&
+                (p.lat() > p1.lat().min(p2.lat())) &&
+                (p.lat() < p1.lat().max(p2.lat()))) {
+                return true;
+            }
+        }
+        return false;
+    }
+}
+#[derive(PartialEq, Clone, Debug)]
+enum PositionPoint {
+    OnBoundary,
+    Inside,
+    Outside,
+}
+
+fn make_position(p: &Point, linestring: &LineString) -> PositionPoint {
+    // Args:
+    // 		linestring : LineString object
+    // 		p		   : Point object
+    // Returns:
+    // 		return a PositionPoint enum.
+
+    let vect = &linestring.0;
+    // LineString without points
+    if vect.is_empty() {
+        return PositionPoint::Outside;
+    }
+    // Point is on linstring
+    if linestring.contains(p) {
+        return PositionPoint::OnBoundary;
+    }
+
+    let mut xints = 0.;
+    let mut crossings = 0;
+    for (p1, p2) in vect.iter().zip(vect[1..].iter()) {
+        if p.lat() > p1.lat().min(p2.lat()) {
+            if p.lat() <= p1.lat().max(p2.lat()) {
+                if p.lng() <= p1.lng().max(p2.lng()) {
+                    if p1.lat() != p2.lat() {
+                        xints = (p.lat() - p1.lat()) * (p2.lng() - p1.lng()) /
+                                (p2.lat() - p1.lat()) + p1.lng();
+                    }
+                    if (p1.lng() == p2.lng()) || (p.lng() <= xints) {
+                        crossings += 1;
+                    }
+                }
+            }
+        }
+    }
+    if crossings % 2 == 1 {
+        PositionPoint::Inside
+    } else {
+        PositionPoint::Outside
+    }
+}
+
+impl Contains<Point> for Polygon {
+    /// See: http://www.ecse.rpi.edu/Homepages/wrf/Research/Short_Notes/pnpoly.html
+    /// 	 http://geospatialpython.com/search
+    ///		?updated-min=2011-01-01T00:00:00-06:00&updated-max=2012-01-01T00:00:00-06:00&max-results=19
+    fn contains(&self, p: &Point) -> bool {
+        let linestring = &self.0;
+        let value = make_position(p, linestring);
+        if value == PositionPoint::OnBoundary || value == PositionPoint::Outside {
+            false
+        } else {
+            let vect_linestring = &self.1;
+            for ls in vect_linestring.iter() {
+                let tmp = make_position(p, ls);
+                if tmp == PositionPoint::OnBoundary || tmp == PositionPoint::Inside {
+                    return false;
+                }
+            }
+            true
+        }
+    }
+}
+
+impl Contains<Point> for MultiPolygon {
+    fn contains(&self, p: &Point) -> bool {
+        let multipolygon = &self.0;
```

Comment:
minor: you can rewrite it with the `any` function
`multipolygon.iter().any(|poly| poly.contains(p))`